### PR TITLE
Refactoring XML IIDM serialization classes

### PR DESCRIPTION
**Pull request by miovd**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No but it has been discussed in PR about replacing DanglingLine by HalfLine

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Refactoring (XIIDM serialization) to be less strict for complex equipments (shunt compensators and in the future, tie lines)

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [x] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*
Might create issues in extensions serialization elsewhere

### CODE REVIEW

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/AliasesXml.java [hunk 1]
```diff
@@ -45,6 +47,16 @@ public static void read(Identifiable<?> identifiable, NetworkXmlReaderContext co
...
     }
 
+    public static <T extends Identifiable> void read(List<Consumer<T>> toApply, NetworkXmlReaderContext context) throws XMLStreamException {
+        if (!context.getReader().getLocalName().equals(ALIAS)) {
+            throw new IllegalStateException();
+        }
+        String[] aliasType = new String[1];
+        IidmXmlUtil.runFromMinimumVersion(IidmXmlVersion.V_1_4, context, () -> aliasType[0] = context.getReader().getAttributeValue(null, "type"));
+        String alias = context.getAnonymizer().deanonymizeString(context.getReader().getElementText());
+        toApply.add(identifiable -> identifiable.addAlias(alias, aliasType[0]));
```
Comment (flo-dup) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/AliasesXml.java:57]: If we don't factorize `getSubElements`, we could factorize this with a method returning `identifiable -> identifiable.addAlias(alias, aliasType[0])`, which is then applied or not.

**Review by flo-dup — CHANGES_REQUESTED** [review]

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/AbstractComplexIdentifiableXml.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    protected void readElement(String id, A adder, List<Consumer<T>> toApply, NetworkXmlReaderContext context) throws XMLStreamException {
+        readRootElementAttributes(adder, toApply, context);
+        readSubElements(id, adder, toApply, context);
+    }
+
+    protected void apply(T identifiable, List<Consumer<T>> toApply) {
+        toApply.forEach(consumer -> consumer.accept(identifiable));
+    }
+
+    public final void read(Supplier<A> createAdder, Function<A, T> create, NetworkXmlReaderContext context) throws XMLStreamException {
```
[the remark this answers is not in the document]
Comment (flo-dup) [reply iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/AbstractComplexIdentifiableXml.java:55]: I forgot the `Function<A, T> create` which is needed. We could do the same, that is, an abstract method which returns the adder. Or, better! in fact we could write `adder.add()` as I'd really like to add `add` methods in the `Adder`s

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/AbstractComplexIdentifiableXml.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+ */
+abstract class AbstractComplexIdentifiableXml<T extends Identifiable, A extends IdentifiableAdder<A>, P extends Identifiable> extends AbstractIdentifiableXml<T, A, P> {
+
+    protected void readUntilEndRootElement(XMLStreamReader reader, XmlUtil.XmlEventHandler eventHandler) throws XMLStreamException {
+        XmlUtil.readUntilEndElement(getRootElementName(), reader, eventHandler);
+    }
+
+    protected abstract void readRootElementAttributes(A adder, List<Consumer<T>> toApply, NetworkXmlReaderContext context);
+
+    protected void readSubElements(String id, A adder, List<Consumer<T>> toApply, NetworkXmlReaderContext context) throws XMLStreamException {
```
[the remark this answers is not in the document]
Comment (miovd) [reply iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/AbstractComplexIdentifiableXml.java:35]: I named the methods the same but made `readSubElements` abstract

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/AbstractComplexIdentifiableXml.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (miovd) [reply iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/AbstractComplexIdentifiableXml.java:55]: In this PR, if your change in IIDM network is not passed, I will create an abstract method to add but I'm keeping it in mind!

**Review by flo-dup — APPROVED** [review]

**Pull request merged**